pkg/ebpf/debugfs.go

```diff
@@ -31,31 +31,31 @@ func GetProbeStats() map[string]int64 {
 
 	res := make(map[string]int64, 2*len(m))
 	for event, st := range m {
-		res[fmt.Sprintf("%s_hits", event)] = st.hits
-		res[fmt.Sprintf("%s_misses", event)] = st.miss
+		res[fmt.Sprintf("%s_hits", event)] = st.Hits
+		res[fmt.Sprintf("%s_misses", event)] = st.Miss
 	}
 
 	return res
 }
 
-// getProbeTotals returns the total number of kprobes triggered or missed by reading the kprobe_profile file
-func getProbeTotals() kprobeStats {
-	stats := kprobeStats{}
+// GetProbeTotals returns the total number of kprobes triggered or missed by reading the kprobe_profile file
+func GetProbeTotals() KProbeStats {
+	stats := KProbeStats{}
 	m, err := readKprobeProfile(KprobeProfile)
 	if err != nil {
 		log.Debugf("error retrieving probe stats: %s", err)
 		return stats
 	}
 
 	for _, st := range m {
-		stats.hits += st.hits
-		stats.miss += st.miss
+		stats.Hits += st.Hits
+		stats.Miss += st.Miss
 	}
 	return stats
 }
 
 // readKprobeProfile reads a /sys/kernel/debug/tracing/kprobe_profile file and returns a map of probe -> stats
-func readKprobeProfile(path string) (map[string]kprobeStats, error) {
+func readKprobeProfile(path string) (map[string]KProbeStats, error) {
```

Comment:
Slightly bugs me that you the "p" in readKprobeProfile isn't capitalized but it is in KProbeStats ;)

Reply:
Fair. I looked it up and "Kprobe" is canonical, so I'm fixing it.